File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        boolean mode = true;
        String[] ArraysStr = code.split("");
        for (int i=0; i<ArraysStr.length; i++) {
            if (ArraysStr[i].equals("1")) {
                mode = !mode;
            }
            else if(mode==true){
                if(i%2==0){
                    answer = answer + ArraysStr[i];
                }
            }
            else if(mode==false){
                if(i%2==1){
                    answer = answer + ArraysStr[i];
                }
            }
        }
        if(answer.equals("")){
            return "EMPTY";
        }
        return answer;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.go
func solution(code string) string {
	ret := ""
	mode := 0

	for idx := 0; idx < len(code); idx++ {
		if mode == 0 {
			if code[idx] != '1' {
				if idx%2 == 0 {
					ret += string(code[idx])
				}
			} else {
				mode = 1
			}
		} else {
			if code[idx] != '1' {
				if idx%2 != 0 {
					ret += string(code[idx])
				}
			} else {
				mode = 0
			}
		}
	}

	if ret == "" {
		return "EMPTY"
	}

	return ret
}